vm/src/org/mozartoz/truffle/nodes/call/CallMethodNode.java

```diff
@@ -33,11 +43,36 @@ public static CallMethodNode create() {
 
 	static final RecordFactory OTHERWISE_MESSAGE_FACTORY = Arity.build("otherwise", 1L).createFactory();
 
-	@Specialization
+	@Specialization(guards = {
+			"cachedProc != null",
+			"self.getClazz() == cachedClazz",
+			"cachedLabel.equals(labelFromArgs(args))",
+	}, limit = "optional(3)")
+	protected Object callClassLabelIdentity(VirtualFrame frame, OzObject self, Object[] args,
+			@Cached("self.getClazz()") DynamicObject cachedClazz,
+			@Cached("labelFromArgs(args)") Object cachedLabel,
+			@Cached("methodFromArgs(self, args)") OzProc cachedProc,
+			@Cached("createDirectCallNode(cachedProc.callTarget)") DirectCallNode callNode) {
+		assert args.length == 1;
+		Object[] arguments = new Object[] { self, args[0] };
+		return callNode.call(frame, OzArguments.pack(cachedProc.declarationFrame, arguments));
+	}
+
+	@Specialization(guards = {
+			"cachedProc != null",
+			"cachedProc == methodFromArgs(self, args)",
+	}, contains = "callClassLabelIdentity", limit = "optional(3)")
+	protected Object callProcIdentity(VirtualFrame frame, OzObject self, Object[] args,
```

Comment:
This specialization seems a bit special-purpose, as it will still perform the lookup all the time. What is the scenario benefiting from it?